pkg/trace/agent/agent.go

```diff
@@ -242,13 +247,8 @@ func (a *Agent) Process(p *api.Payload) {
 
 		// Extra sanitization steps of the trace.
 		for _, span := range chunk.Spans {
-			for k, v := range a.conf.GlobalTags {
-				if k == tagOrigin {
-					chunk.Origin = v
-				} else {
-					traceutil.SetMeta(span, k, v)
-				}
-			}
+			spanMeta := span.GetMeta()
+			a.setGlobalTags(spanMeta, chunk, span)
```

Comment:
```suggestion
			if span.Meta[tagInferredSpan] != "self" {
				// we don't set global tags on Lambda spans with tagInferredSpan == "self"
				for k, v := range a.conf.GlobalTags {
					if k == tagOrigin {
						chunk.Origin = v
					} else {
						traceutil.SetMeta(span, k, v)
					}
				}
			}
```
I like the use of `GetMeta` but we haven't used it anywhere, so for the sake of consistency, can we not use it please? Also, I don't think we need to check the second argument to the map read, since an empty string will be returned always: even if the map is `nil` or the key doesn't exist.

Lastly, I don't think we need a separate method, because it's not used anywhere else.

Reply:
Makes sense. I extracted the method because I thought that was going to be the best way to unit test the new behavior, but then I figured out how `TestProcess` works. Done.

Reply:
Thanks. Appreciate it.